src/algebra/group/pi.lean

```diff
@@ -47,6 +60,18 @@ instance group [∀ i, group $ f i] : group (Π i : I, f i) :=
 by refine_struct { one := (1 : Π i, f i), mul := (*), inv := has_inv.inv, div := has_div.div, .. };
   tactic.pi_instance_derive_field
 
+lemma single_neg [∀ i, add_group $ f i] [decidable_eq I] (i : I) (x : f i) :
```

Comment:
This proof would be a one liner if `add_monoid_hom.single` were already available, right?

Reply:
I hadn't thought of that. Yes.